xwiki-pro-macros-test/xwiki-pro-macros-test-docker/src/test/it/com/xwiki/pro/test/ui/GenericMacrosIT.java

```diff
@@ -76,61 +101,582 @@
+        assertTrue(status3.hasColor("yellow"));
+        assertFalse(status3.isSubtle());
+    }
+
+    @Test
+    @Order(4)
+    void tagListMacroTest(TestUtils setup)
+    {
+        createPagesWithTags(setup);
+        setup.createPage(pageWithTagListMacros, createContent("taglist-macros.vm"));
+
+        TagListMacroPage page = new TagListMacroPage();
+        TagListMacro tagList1 = page.getMacro(0);
+        TagListMacro tagList2 = page.getMacro(1);
+        TagListMacro tagList3 = page.getMacro(2);
+
+        // There should be 3 tagList macros.
+        assertEquals(3, page.getMacroCount());
+
+        // Checks the 1st tagList macro, which should contain all the tags.
+        List<String> expectedTitles1 = Arrays.asList("A-B", "G");
+        assertEquals(expectedTitles1, tagList1.getGlossaryTitles());
+
+        List<String> expectedTagNames1 = Arrays.asList("alpha", "beta", "gamma");
+        assertEquals(expectedTagNames1, tagList1.getTagNames());
+
+        // Checks the links associated with the tags.
+        for (String i : expectedTagNames1) {
+            assertEquals("a", tagList1.getHtmlTagForTagName(i));
+        }
+
+        // Checks the 2nd tagList macro, which excludes the "beta" tag.
+        List<String> expectedTitles2 = Arrays.asList("A-G");
+        assertEquals(expectedTitles2, tagList2.getGlossaryTitles());
+        List<String> expectedTagNames2 = Arrays.asList("alpha", "gamma");
+        assertEquals(expectedTagNames2, tagList2.getTagNames());
+        for (String i : expectedTagNames2) {
+            assertEquals("a", tagList2.getHtmlTagForTagName(i));
+        }
+
+        // Checks the 3rd tagList macro, which targets specific spaces and excludes the "gamma" tag.
+        List<String> expectedTitles3 = Arrays.asList("A-X", "Y-Z");
+        assertEquals(expectedTitles3, tagList3.getGlossaryTitles());
+
+        List<String> expectedTagNames3 = Arrays.asList("alpha", "beta", "testTag", "x", "y", "z");
+        assertEquals(expectedTagNames3, tagList3.getTagNames());
+
+        for (String i : expectedTagNames3) {
+            assertEquals("a", tagList3.getHtmlTagForTagName(i));
+        }
+    }
+
+    @Test
+    @Order(5)
+    void userProfileMacroTest(TestUtils setup, TestReference testReference)
+    {
+        setup.createPage(testReference, createContent("userprofile-macros.vm"), "UserProfileTest");
+
+        UserProfileMacroPage page = new UserProfileMacroPage();
+
+        UserProfileMacro user1 = page.getMacro(0);
+        UserProfileMacro user2 = page.getMacro(1);
+        UserProfileMacro user3 = page.getMacro(2);
+
+        // There should be 3 userProfile macros.
+        assertEquals(3, page.getMacroCount());
+
+        // Checks the 1st userProfile macro, with the properties: company, email, phone, address.
+        assertTrue(user1.hasLinkImage("UserTest"));
+        assertTrue(user1.imageHasTitle("UserTest"));
```

Comment:
I would combine this 2 maybe in `hasAvatarWithLink`? And you would do both checks in there
There are more or less the same element and you check similar things